src/DomainNetMap.cpp

```diff
@@ -0,0 +1,265 @@
+        return m_nn_factory->createDenseLayer(weights, biases);
+    }
+
+    TensorOneD DomainNetMapImp::json_to_TensorOneD(const json11::Json &obj) const {
+        if (!obj.is_array()) {
+            throw Exception("Neural network weights contains non-array-type.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        if (obj.array_items().empty()) {
+            throw Exception("Empty array is invalid for neural network weights.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        std::size_t vec_size = obj.array_items().size();
+        std::vector<double> vals(vec_size);
+
+        for (std::size_t idx = 0; idx < vec_size; ++idx) {
+            if (!obj[idx].is_number()) {
+                throw Exception("Non-numeric type found in neural network weights.\n",
+                                       GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+            vals[idx] = obj[idx].number_value();
+        }
+
+        return m_nn_factory->createTensorOneD(vals);
+    }
+
+    TensorTwoD DomainNetMapImp::json_to_TensorTwoD(const json11::Json &obj) const {
+        if (!obj.is_array()){
+            throw Exception("Neural network weights is non-array-type.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        if (obj.array_items().size() == 0) {
+            throw Exception("Empty array is invalid for neural network weights.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        std::vector<std::vector<double> > vals;
+        for (std::size_t ridx = 0; ridx < obj.array_items().size(); ++ridx) {
+            if(!obj[ridx].is_array()) {
+                throw Exception("Neural network weights is non-array-type.\n",
+                                       GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+            std::size_t vec_size = obj[ridx].array_items().size();
+            std::vector<double> row(vec_size);
+
+            for (std::size_t cidx = 0; cidx < vec_size; ++cidx) {
+                if (!obj[ridx][cidx].is_number()) {
+                    throw Exception("Non-numeric type found in neural network weights.\n",
+                                           GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+                }
+                row[cidx] = obj[ridx][cidx].number_value();
+            }
+
+            vals.push_back(row);
+        }
+
+        return m_nn_factory->createTensorTwoD(vals);
+    }
+
+    void DomainNetMapImp::sample()
+    {
+        std::vector<double> xs;
+
+        // Sample latest signal values
+        for (std::size_t i=0; i<m_signal_inputs.size(); i++) {
+            m_signal_inputs[i].signal = m_platform_io.sample(m_signal_inputs[i].batch_idx);
+            xs.push_back(m_signal_inputs[i].signal);
+        }
+        for (std::size_t i=0; i<m_delta_inputs.size(); i++) {
```

Comment:
```suggestion
        for (const &auto inputs : m_delta_inputs) {
```

Modify lines below accordingly